### DISCUSSION TITLE
UART Lilygo to D1 Mini (RX-TX)

### DISCUSSION THREAD

**Discussion by javrolijk** [timmbogner/Farm-Data-Relay-System#97] [Q&A] [answered]
Hi folks,

I'm trying to send data received from a Lora sensor on a Lilygo/TTGO ESP32 Lora module to a D1 Mini module through serial (UART). Both modules can be seen below.  I've connected the Lora TXD port to the RX port on the D1, but when I do so, the Lora module stops sending data. When the wire is removed, it starts sending data. I have #define FDRS_DEBUG enabled.
I'm unsure if I have the wire connected to the Lilygo ESP32 Lora module correctly. Can anyone provide a hand?

![image](https://user-images.githubusercontent.com/70583691/185529995-04df7d5c-7d21-4c59-99fe-85b81efd3f29.png)
![image](https://user-images.githubusercontent.com/70583691/185529438-94802afe-c6e2-473a-ad82-4e8812e07e3a.png)

Here's how I have the UART_Gateway setup:
`//  FARM DATA RELAY SYSTEM
//
//  GATEWAY 2.000 Configuration

#include <fdrs_globals.h>
#define FDRS_DEBUG     //Enable USB-Serial debugging

#define UNIT_MAC     0x04  // The address of this gateway

// Actions -- Define what happens when a packet arrives at each interface:
// Current function options are: sendESPNOW(MAC), sendSerial(), sendMQTT(),
//  bufferLoRa(interface), bufferESPNOW(interface), bufferSerial(), and bufferMQTT().

#define ESPNOWG_ACT
#define SERIAL_ACT         
#define MQTT_ACT          
#define LORAG_ACT      sendSerial();

#define USE_LORA
//#define USE_WIFI    //Used only for MQTT gateway

// Peer addresses
#define ESPNOW1_PEER  0x0E  // ESPNOW1 Address 
#define ESPNOW2_PEER  0x0F  // ESPNOW2 Address
#define LORA1_PEER    0x0E  // LoRa1 Address
#define LORA2_PEER    0x0F  // LoRa2 Address

// Peer Actions
#define ESPNOW1_ACT    
#define ESPNOW2_ACT                    
#define LORA1_ACT 
#define LORA2_ACT 

//Pins for UART data interface (ESP32 only)
#define RXD2 14
#define TXD2 15
//Logging settings  --  Logging will occur when MQTT is disconnected
//#define USE_SD_LOG        //Enable SD-card logging
//#define USE_FS_LOG        //Enable filesystem (flash) logging
#define LOGBUF_DELAY 10000  // Log Buffer Delay - in milliseconds
#define SD_SS 0             //SD card CS pin (Use different pins for LoRa and SD)
#define SD_FILENAME "fdrs_log.csv"
#define FS_FILENAME "fdrs_log.csv"

// SPI Configuration -- Needed only on Boards with multiple SPI interfaces like the ESP32

#define SPI_SCK 5
#define SPI_MISO 19
#define SPI_MOSI 27

// LoRa Configuration -- Needed only if using LoRa
#define LORA_SS 18
#define LORA_RST 23
#define LORA_DIO0 26

//433E6 for Asia
//866E6 for Europe
//915E6 for North America
#define LORA_BAND 915E6
#define LORA_SF 7

// Buffer Delays - in milliseconds - Uncomment to enable any buffer

//#define ESPNOW1_DELAY  0
//#define ESPNOW2_DELAY  0
//#define ESPNOWG_DELAY  0
//#define SERIAL_DELAY   0
//#define MQTT_DELAY     0
//#define LORAG_DELAY    1000
//#define LORA1_DELAY    1000
//#define LORA2_DELAY    1000

// FastLED -- Not yet fully implemented
//#define USE_LED
#define LED_PIN    32
#define NUM_LEDS    4

// WiFi and MQTT Credentials -- Needed for MQTT only if "fdrs_globals.h" is not included
#define WIFI_SSID   "Your SSID"  
#define WIFI_PASS   "Your Password"
#define MQTT_ADDR   "192.168.0.8"
#define MQTT_PORT   1883 // Default MQTT port is 1883

//#define MQTT_AUTH   //Enable MQTT authentication 
#define MQTT_USER   "Your MQTT Username"
#define MQTT_PASS   "Your MQTT Password"

// MQTT Topics
#define TOPIC_DATA "fdrs/data"
#define TOPIC_STATUS "fdrs/status"
#define TOPIC_COMMAND "fdrs/command" `

**Comment by aviateur17** [comment]
When you say it stops sending data are you talking about serial data or LoRa data?   How do you know it stops sending data?  What shows you that?  Do you have a ground wire connected between the two modules?  What difference does making a ground connection between both modules make?

**Reply by javrolijk** [reply]
Thanks for the prompt response. 
I've connected a ground wire between the two boards as well. It just stops the data flow when the TDX to RX wire is connected.

![Screenshot 2022-08-18 232623_1](https://user-images.githubusercontent.com/70583691/185538469-214c2a1b-df48-4b6a-8eab-9598c818356b.png)

On the other side, the MQTT Gateway does receive the serial data.
![image](https://user-images.githubusercontent.com/70583691/185539317-10ed3a1e-4879-4cb2-894c-8261c932cf68.png)

**Comment by aviateur17** [comment]
Any difference if you connect another wire between Rx on the LoRa board and TX on the MQTT board?   If you power down the MQTT board even when the TX -> RX wire is present I assume that the Incoming LoRa and Sending serial messages come back?  Seems like LoRa packets are not being received when you have the TX -> RX wire connected.    What does the startup of the LoRa board look like when the MQTT board is already powered up and you then power up the LoRa board?  What messages do you see on startup?

**Comment by timmbogner** [comment, marked as answer]
I think you're using the wrong Tx pin for data. Your config defines pin 15 as the Tx for data, so try switching to that for starters.

On ESP8266 the serial port is singular and has fixed pins, but on ESP32 we can define a second serial port on which to send the data.

**Reply by javrolijk** [reply]
Tim,
That's exactly the answer. It seems confusing the TXD and RDX on the Lilygo/TTGO ESP32 Lora module. How would those ports be used for the transfer and receive?
